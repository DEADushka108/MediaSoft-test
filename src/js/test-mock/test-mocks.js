export * from './const/const';
export * from './ships/ships';
export * from './store/store';
