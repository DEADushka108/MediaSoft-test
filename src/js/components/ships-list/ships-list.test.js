import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {shipsList, store, MAX_CARD_COUNT} from '../../test-mock/test-mocks';
import ShipsList from './ships-list';
import {MemoryRouter} from 'react-router-dom';

it(`Should ShipsList component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <ShipsList
            ships={shipsList}
            count={MAX_CARD_COUNT}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
