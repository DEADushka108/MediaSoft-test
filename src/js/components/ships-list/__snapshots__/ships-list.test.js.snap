// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ShipsList component render correctly 1`] = `
<div
  className="catalog__ships-list"
>
  <article
    className="small-card"
  >
    <div
      className="small-card__image"
      onClick={[Function]}
    >
      <img
        className="small-card__img"
        src="./img/content/battlestar-galactica-preview.png"
      />
    </div>
    <div
      className="small-card__wrapper"
    >
      <h3
        className="small-card__name"
      >
        <a
          className="small-card__link"
          href="/card/3"
          onClick={[Function]}
        >
          Battlestar Galactica
        </a>
      </h3>
      <p
        className="small-card__price"
      >
        234122
        $
      </p>
    </div>
    <button
      className="small-card__button"
      onClick={[Function]}
      type="button"
    >
      In cart
       
    </button>
  </article>
  <article
    className="small-card"
  >
    <div
      className="small-card__image"
      onClick={[Function]}
    >
      <img
        className="small-card__img"
        src="./img/content/borgs-cube-preview.jpg"
      />
    </div>
    <div
      className="small-card__wrapper"
    >
      <h3
        className="small-card__name"
      >
        <a
          className="small-card__link"
          href="/card/4"
          onClick={[Function]}
        >
          Borg's Cube
        </a>
      </h3>
      <p
        className="small-card__price"
      >
        1002000
        $
      </p>
    </div>
    <button
      className="small-card__button"
      onClick={[Function]}
      type="button"
    >
      In cart
       
    </button>
  </article>
  <article
    className="small-card"
  >
    <div
      className="small-card__image"
      onClick={[Function]}
    >
      <img
        className="small-card__img"
        src="./img/content/death-star-preview.png"
      />
    </div>
    <div
      className="small-card__wrapper"
    >
      <h3
        className="small-card__name"
      >
        <a
          className="small-card__link"
          href="/card/5"
          onClick={[Function]}
        >
          Death Star
        </a>
      </h3>
      <p
        className="small-card__price"
      >
        1000000
        $
      </p>
    </div>
    <button
      className="small-card__button"
      onClick={[Function]}
      type="button"
    >
      Add
       
    </button>
  </article>
  <article
    className="small-card"
  >
    <div
      className="small-card__image"
      onClick={[Function]}
    >
      <img
        className="small-card__img"
        src="./img/content/eagle-5-preview.jpg"
      />
    </div>
    <div
      className="small-card__wrapper"
    >
      <h3
        className="small-card__name"
      >
        <a
          className="small-card__link"
          href="/card/6"
          onClick={[Function]}
        >
          Eagle-5
        </a>
      </h3>
      <p
        className="small-card__price"
      >
        100
        $
      </p>
    </div>
    <button
      className="small-card__button"
      onClick={[Function]}
      type="button"
    >
      Add
       
    </button>
  </article>
  <article
    className="small-card"
  >
    <div
      className="small-card__image"
      onClick={[Function]}
    >
      <img
        className="small-card__img"
        src="./img/content/flying-dishes-preview.jpg"
      />
    </div>
    <div
      className="small-card__wrapper"
    >
      <h3
        className="small-card__name"
      >
        <a
          className="small-card__link"
          href="/card/7"
          onClick={[Function]}
        >
          Flying Dish
        </a>
      </h3>
      <p
        className="small-card__price"
      >
        3400000
        $
      </p>
    </div>
    <button
      className="small-card__button"
      onClick={[Function]}
      type="button"
    >
      Add
       
    </button>
  </article>
  <article
    className="small-card"
  >
    <div
      className="small-card__image"
      onClick={[Function]}
    >
      <img
        className="small-card__img"
        src="./img/content/golden-heart-preview.png"
      />
    </div>
    <div
      className="small-card__wrapper"
    >
      <h3
        className="small-card__name"
      >
        <a
          className="small-card__link"
          href="/card/8"
          onClick={[Function]}
        >
          Heart of Gold
        </a>
      </h3>
      <p
        className="small-card__price"
      >
        100000000
        $
      </p>
    </div>
    <button
      className="small-card__button"
      onClick={[Function]}
      type="button"
    >
      Add
       
    </button>
  </article>
</div>
`;
